packages/react-utilities/src/hooks/useOnClickOutside.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import { useEventCallback } from './useEventCallback';
+
+export const useOnClickOutside = (options: {
+  element?: Node | Window | Document;
+  refs: React.MutableRefObject<HTMLElement>[];
+  callback: (ev: Event) => void;
+}) => {
+  const { refs, callback, element = document } = options;
+  const listener = useEventCallback((ev: Event) => {
```

Comment:
```suggestion
  element?: Document;
  refs: React.MutableRefObject<HTMLElement>[];
  callback: (ev: MouseEvent | TouchEvent) => void;
}) => {
  const { refs, callback, element = document } = options;
  const listener = useEventCallback((ev: MouseEvent | TouchEvent) => {
```

I suggest to restrict `element` to `Document` only as it's a single usage currently. It will also allow us to use proper typings for `ev`